src/v/cluster/node/local_monitor.cc

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/node/local_monitor.h"
+
+#include "cluster/node/types.h"
+#include "config/node_config.h"
+#include "model/timestamp.h"
+#include "version.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/file.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/reactor.hh>
+
+#include <chrono>
+#include <seastarx.h>
+
+namespace cluster::node {
+
+ss::future<> local_monitor::update_state() {
+    auto disks = co_await get_disks();
+    auto vers = application_version((std::string)redpanda_version());
+    auto ts = model::timestamp::now();
+    auto uptime = std::chrono::duration_cast<std::chrono::milliseconds>(
+      ss::engine().uptime());
+
+    _state = {
+      .redpanda_version = vers,
+      .uptime = uptime,
+      .timestamp = ts,
+      .disks = disks,
+    };
+    co_return;
+}
+
+const local_state& local_monitor::get_state_cached() { return _state; }
+
+ss::future<std::vector<disk>> local_monitor::get_disks() {
```

Comment:
Good catch making this async.

There are a bunch of other places we use the synchronous system call via make_self_broker.  I think/hope they're all just at startup time though.

Reply:
I thought I knew about just a few places on start-up with synchronous system calls. I don't see any via make_self_broker. where are you seeing them?